woocommerce/woocommerce-ios
# Dismiss all notes: fix some notes are not deleted

Closes: #6282 
- [x] ⚠️ Make sure https://github.com/woocommerce/woocommerce-ios/pull/6275 is merged before merging ⚠️ 


### Description


While testing Paolo's PR that dismisses all notes, we can reproduce an issue where some notes still appear after `delete/all` API request. After debugging with Jurassic sites and checking the [backend code](https://github.com/woocommerce/woocommerce-admin/blob/b58e41d8778448b55b9d8e0e5d26813034fda4e5/src/Notes/Notes.php#L186-L219), the reason is that the backend first **fetches the notes with the API params**, then "soft deletes" them (setting is_deleted=true) instead of deleting all notes in the database. On a brand new Jurassic Ninja site, there are a few other notes that we don't fetch since we specify parameters to filter notes by type and status). Since we don't pass any params in the `delete/all` request, it fetches and soft deletes all notes including types and statuses that we don't fetch and display. Thus, the first 25 notes (default page size) that the backend deletes don't include all the notes that were previously fetched and these leftover notes stay undeleted.

This PR fixed the issue by passing the same order, type, and status parameters as the GET request, with a page size that includes all synced notes. This means that if the user hasn't seen/loaded all the notes in the database with infinite scroll, they could still see notes after revisiting the Inbox page. I think this is fine because we only want to delete/dismiss notes that the user has seen.

### Testing instructions


It's the easiest to test on a new Jurassic Ninja site.

* Launch the app
* Go to the Menu tab
* Tap on "Inbox" --> after syncing, some notes should appear
* Scroll down until there are no more notes to load
* Tap the ellipsis menu on the navigation bar
* Tap "Dismiss All" and confirm the alert --> the notes should disappear and the empty state is shown
* Go back to the Menu tab
* Tap on "Inbox" --> after syncing, no notes should appear and the empty state is shown

Tip: you can undo the deletion by making `POST /wc-analytics/admin/notes/update` with body like (`40` is the max note ID that was synced):

```
{"noteIds":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40],"is_deleted":0}
```
I first copied the undo request as URL from core in Chrome debugger Network tab, then imported to Postman and modified the body.


---
